x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/automated_curation.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useParams } from 'react-router-dom';
+
+import { useValues, useActions } from 'kea';
+
+import { EuiSpacer, EuiButton, EuiBadge } from '@elastic/eui';
+
+import { AppSearchPageTemplate } from '../../layout';
+import { AutomatedIcon } from '../components/automated_icon';
+import {
+  AUTOMATED_LABEL,
+  COVERT_TO_MANUAL_BUTTON_LABEL,
+  CONVERT_TO_MANUAL_CONFIRMATION,
+} from '../constants';
+import { getCurationsBreadcrumbs } from '../utils';
+
+import { CurationLogic } from './curation_logic';
+import { PromotedDocuments, OrganicDocuments } from './documents';
+
+export const AutomatedCuration: React.FC = () => {
+  const { curationId } = useParams() as { curationId: string };
```

Comment:
I suspect you can do something like this instead to avoid casting.

```
const { curationId } = useParams<{ curationId: string }>();
```

Reply:
Thank you! I did this in 9930246